File: Java/AutomationFramework/src/com/app/pages/DashboardPage.java
package com.app.pages;

import com.app.helpers.actions.DriverActions;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import java.util.List;

public class DashboardPage{
    private static BasePage Base_Page;
    private static DashboardPage Dashboard_Page;
    public static DriverActions driverActions;
    public static WebDriver driver;

    private DashboardPage() {
    }

    public static DashboardPage GetDashboardPage() {
        if(Dashboard_Page == null) {
            synchronized (LoginPage.class) {
                if(Dashboard_Page == null)
                    Dashboard_Page = new DashboardPage();
            }
        }
        Base_Page = BasePage.GetBasePage();
        driverActions = Base_Page.driverActions;
        return Dashboard_Page;
    }

    public DashboardPage Init_Elements() {

        PageFactory.initElements(Base_Page.driver, this);
        return this;
    }

    @FindBy(css = ".oxd-brand-banner img")
    private WebElement Logo;
    @FindBy(css = ".oxd-topbar h6")
    private WebElement Header;
    @FindBy(css = ".oxd-main-menu-item span")
    private List<WebElement> MenuList;
    @FindBy(xpath = "//a[@class='oxd-main-menu-item active']/span")
    private WebElement ActiveMenu;
    @FindBy(xpath = "//span[text()='Leave']")
    private WebElement LeaveMenu;

    public boolean IsLogoPresent()
    {
        boolean isPresent = driverActions.GetPresenceOfElement(Logo);
        return isPresent;
    }

    public DashboardPage WaitForPageLoad()
    {
        driverActions.WaitUntilElementVisibleAndClickable(Logo);
        return this;
    }

    public String GetHeaderText()
    {
        String header = driverActions.GetText(Header);
        return header;
    }

    public String  GetActiveMenuText()
    {
        String activeMenu = driverActions.GetText(ActiveMenu);
        return activeMenu;
    }

    public List<String> GetMenuList()
    {
        List<String> menuList = driverActions.GetText(MenuList);
        return menuList;
    }

    public boolean GetPresenceMenu(String menu)
    {
        List<String> menuList = driverActions.GetText(MenuList);
        boolean isPresent = menuList.contains(menu);
        return isPresent;
    }

    public LeavePage GoToLeavePage()
    {
        driverActions.WaitUntilElementVisibleAndClickable(LeaveMenu);
        driverActions.mouseActions.ClickElement(LeaveMenu);
        return LeavePage.GetLeavePage()
                .Init_Elements();
    }
}

File: DotNet/Framework/AutomationFramework/Pages/DashboardPage.cs
using Helpers.Actions;
using OpenQA.Selenium;

namespace Pages
{
    public sealed class DashboardPage
    {
        private static Lazy<DashboardPage> GetDashboardPage = new Lazy<DashboardPage>(() => new DashboardPage());
        private static BasePage Base_Page;
        private static DriverActions driverActions;
        public static DashboardPage Dashboard_Page { get 
            {
                Base_Page = BasePage.Base_Page;
                driverActions = Base_Page.driverActions;
                return GetDashboardPage.Value; 
            } 
        }

        public DashboardPage()
        {
        }

        #region Locators

        private By Logo = By.CssSelector(".oxd-brand-banner img");
        private By Header = By.CssSelector(".oxd-topbar h6");
        private By MenuList = By.CssSelector(".oxd-main-menu-item span");
        private By ActiveMenu = By.XPath("//a[@class='oxd-main-menu-item active']/span");
        private By LeaveMenu = By.XPath("//span[text()='Leave']");

        #endregion

        #region Actions

        public DashboardPage IsLogoPresent(out bool isPresent)
        {
            isPresent = driverActions.GetPresenceOfElement(Logo);
            return this;
        }

        public DashboardPage WaitForPageLoad()
        {
            driverActions.WaitUntilElementClickable(Logo);
            return this;
        }

        public DashboardPage GetHeaderText(out string header)
        {
            header = driverActions.GetText(Header);
            return this;
        }

        public DashboardPage GetActiveMenuText(out string activeMenu)
        {
            activeMenu = driverActions.GetText(ActiveMenu);
            return this;
        }

        public DashboardPage GetMenuList(out List<string> menuList)
        {
            driverActions.GetText(MenuList, out menuList);
            return this;
        }

        public DashboardPage GetPresenceMenu(string menu, out bool isPresent)
        {
            driverActions.GetText(MenuList, out List<string> menuList);
            isPresent = menuList.Contains(menu);
            return this;
        }

        public LeavePage GoToLeavePage()
        {
            driverActions.WaitUntilElementClickable(LeaveMenu);
            driverActions.mouseActions.ClickElement(LeaveMenu);
            return LeavePage.Leave_Page;
        }

        #endregion
    }
}
